vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -803,6 +808,13 @@ void AdsImpl::OnDidServeAdNotification(const AdNotificationInfo& ad) {
 void AdsImpl::OnAdNotificationViewed(const AdNotificationInfo& ad) {
   account_->Deposit(ad.creative_instance_id, ad.type,
                     ConfirmationType::kViewed);
+
+  auto impression_served_at_covariate_log_entry =
+      std::make_unique<ImpressionServedAtCovariateLogEntry>();
+  impression_served_at_covariate_log_entry->SetLastImpressionAt(
+      base::Time::Now());
+  CovariateLogs::Get()->SetCovariateLogEntry(
```

Comment:
I would prefer not to use global variables so I think it is better to use `covariate_logs_` member here. In this case  `g_covariate_logs` global becomes not used and could be removed.

Reply:
This would also fit in with https://github.com/brave/brave-browser/issues/9164

Reply:
@aseren I took this pattern from ads diagnostics. Could we change along with #9164 or do we wanna change here?

Reply:
Let's change this here. There is no need to create a global variable here.

Reply:
fixed